File: 01/src/iteruj.js
function addN(n) {
	return function(x) { return x+n };
}
function iteruj(n, f) {
   if (n === 0) 
      return function(x) {return x;};
   else
      return function(x) { return iteruj(n-1,f)(f(x)); };
}
add5 = addN(5);
add1 = addN(1);
iteruj(7,add5)(100)
iteruj(7,iteruj(4,add1))(100)

File: 02/src/Iteruj.scala
object HelloWorld {
    def iteruj[A](n:Int, f:(A=>A)) : (A => A) = {
      if (n == 0) 
        return (x:A) => x
      else
        return (x:A) => iteruj(n-1, f)(f(x))
    }
    def main(args: Array[String]) {
      val add5 = (i:Int) => i+5
      println(iteruj(7, (i:Int) => i+1)(100))
      println(iteruj(7, iteruj(4,add5))(100))
    }
}